### ISSUE TITLE
Support for DigitalOcean Spaces

### ISSUE THREAD

**Issue by bschulz87** [craftcms/aws-s3#17]
It would be great if this plugin could also support DigitalOcean Spaces. Currently it's not possible to configure the bucket.

**Comment by jacket** [comment]
Likewise, it would be great this plugin supported [Sirv asset storage](https://sirv.com/). Sirv resizes and optimises images on-the-fly. It uses S3 but on its own S3 infrastructure - not AWS S3.

**Comment by andris-sevcenko** [comment]
This plugin is tailored to work with AWS, not any storage service that uses the same API. Case in point would be using Amazon STS for caching credentials and Amazon CloudFront CDN path invalidation.

It would also increase complexity down the road, as sharing the same API does not mean they share the same behavior. For a reference, see https://cloud.google.com/storage/docs/migrating#methods-comparison.
 
To summarize, adding support for other services would mean dumbing down this plugin and making it harder to maintain down the road.

It shouldn't be hard to make new plugins to support DO Spaces and Sirv, however.
